text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,1207 @@
+   jobType: glue.JobType.ETL,
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_9,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+   description: 'an example pySpark ETL job',
+   numberOfWorkers: 20,
+   role: iam.IRole,
+});
+```
+
+Scala Spark ETL Job Property Interface:
+
+```ts
+ScalaSparkEtlJobProps{
+    /**
+     * Script Code Location (required)
+     * Script to run when the Glue Job executes. Can be uploaded
+     * from the local directory structure using fromAsset
+     * or referenced via S3 location using fromBucket
+     * */
+    script: glue.Code;
+
+    /**
+     * Class name (required for Scala)
+     * Package and class name for the entry point of Glue Job execution for
+     * Java scripts
+     * */
+    className: string;
+
+    /**
+     * Extra Jars S3 URL (optional)
+     * S3 URL where additional jar dependencies are located
+     */
+    extraJarsS3Url?: string[];
+
+    /**
+     * IAM Role (required)
+     * IAM Role to use for Glue Job execution
+     * */
+    role: iam.IRole;
+
+    /**
+     * Name of the Glue Job (optional)
+     * Developer-specified name of the Glue Job
+     * */
+    name?: string;
+
+    /**
+     * Description (optional)
+     * Developer-specified description of the Glue Job
+     * */
+    description?: string;
+
+    /**
+     * Number of Workers (optional)
+     * Number of workers for Glue to use during Job execution
+     * @default 10
+     * */
+    numberOrWorkers?: int;
+
+    /**
+     * Max Concurrent Runs (optional)
+     * The maximum number of runs this Glue Job cna concurrently run
+     * @default 1
+     * */
+    maxConcurrentRuns?: int;
+
+    /**
+     * Default Arguments (optional)
+     * The default arguments for every run of this Glue Job,
+     * specified as name-value pairs.
+     * */
+    defaultArguments?: {[key: string], string }[];
+
+    /**
+     * Connections (optional)
+     * List of connections to use for this Glue Job
+     * */
+    connections?: IConnection[];
+
+    /**
+     * Max Retries (optional)
+     * Maximum number of retry attempts Glue will perform
+     * if the Job fails
+     * @default 0
+     * */
+    maxRetries?: int;
+
+    /**
+     * Timeout (optional)
+     * Timeout for the Glue Job, specified in minutes
+     * @default 2880 (2 days for non-streaming)
+     * */
+    timeout?: int;
+
+    /**
+     * Security Configuration (optional)
+     * Defines the encryption options for the Glue Job
+     * */
+    securityConfiguration?: ISecurityConfiguration;
+
+    /**
+     * Tags (optional)
+     * A list of key:value pairs of tags to apply to this Glue Job resource
+     * */
+    tags?: {[key: string], string }[];
+
+    /**
+     * Glue Version
+     * The version of Glue to use to execute this Job
+     * @default 3.0 for ETL
+     * */
+    glueVersion?: glue.GlueVersion;
+}
+```
+
+pySpark ETL Job Property Interface:
+
+```ts
+pySparkEtlJobProps{
```

Comment:
Please note that the pySpark Jobs should also support the `extraJars` property. Eventhough spark can be interfaced with using Python, libraries from jars can still be loaded/leveraged

Reply:
Addressed this in latest push.